mithril-signer/src/services/epoch_service.rs

```diff
@@ -419,11 +509,106 @@ mod tests {
         assert!(!service
             .is_signer_included_in_current_stake_distribution(
                 party_id,
-                protocol_initializer_not_included
+                &protocol_initializer_not_included
             )
             .unwrap());
     }
 
+    #[test]
+    fn signer_cant_sign_if_no_protocol_initializer_are_stored() {
+        let epoch = Epoch(12);
+        let fixtures = MithrilFixtureBuilder::default().with_signers(10).build();
+        let stake_store = Arc::new(StakeStore::new(Box::new(DumbStoreAdapter::new()), None));
+        let protocol_initializer_store = Arc::new(ProtocolInitializerStore::new(
+            Box::new(DumbStoreAdapter::new()),
+            None,
+        ));
+
+        let epoch_service = MithrilEpochService::new(stake_store, protocol_initializer_store)
+            .set_data_to_default_or_fake(epoch)
+            .alter_data(|data| {
+                data.protocol_initializer = None;
+                data.current_signers = fixtures.signers();
+            });
+
+        let can_sign_current_epoch = epoch_service
+            .can_signer_sign_current_epoch(fixtures.signers()[0].party_id.clone())
+            .unwrap();
+        assert!(!can_sign_current_epoch);
+    }
+
+    #[test]
+    fn signer_cant_sign_if_it_dont_use_stored_protocol_initializer() {
```

Comment:
This test is not very clear: are we trying to test that a protocol initializer created by a different pool can not sign?

Also small typo:
```suggestion
    fn signer_cant_sign_if_it_doesnt_use_stored_protocol_initializer() {
```

Reply:
I renamed the test to `cant_sign_if_stored_protocol_initializer_belong_to_another_party`.

What this test aim to check is the `&& s.verification_key == protocol_initializer.verification_key().into()` part that's in the `is_signer_included_in_current_stake_distribution` private method.